utils.py

```diff
@@ -0,0 +1,194 @@
+            return scenario_params
+    
+    raise ValueError(f"No scenario parameters found for scenario '{scenario_name}' in file '{scenario_file}'.")
+
+
+def extract_reference_band_statistics(scenario_name: str) -> dict:
+    """
+    Loads reference data from a JSON file for a specific scenario.
+
+    Parameters:
+        reference_file (str): The path to the JSON file containing the reference statistics for all scenarios.
+        scenario_name (str): The name of the scenario for which reference data is needed.
+
+    Returns:
+        dict: The reference data for the specified scenario.
+    """
+    reference_file = 'groundtruth_regression_test.json'
+
+    with open(reference_file, 'r') as file:
+        all_reference_data = json.load(file)
+    
+    for scenario_data in all_reference_data:
+        if scenario_data['scenario_name'] == scenario_name:
+            return scenario_data['reference_data']
+    
+    raise ValueError(f"No reference data found for scenario '{scenario_name}' in file '{reference_file}'.")
+    
+
+def compare_band_statistics(output_band_stats: dict, groundtruth_band_dict: dict, tolerance: float) -> None:
+    """
+    Compares the statistics of different bands in the output against the reference data.
+
+    Parameters:
+        output_band_stats (dict): The output dictionary containing band statistics to be compared.
+        groundtruth_band_dict (dict): The reference dictionary containing expected band statistics.
+        tolerance (float): Tolerance value for comparing values.
+
+    Returns:
+        None
+    """
+    for output_band_name, output_band_stats in output_band_stats.items():
+        if output_band_name not in groundtruth_band_dict:
+            print(f"Warning: Band '{output_band_name}' not found in reference.")
+            continue
+
+        gt_band_stats = groundtruth_band_dict[output_band_name]
+        for stat, gt_value in gt_band_stats.items():
+            if stat not in output_band_stats:
+                print(f"Warning: Statistic '{stat}' not found for band '{output_band_name}' in output.")
+                continue
+
+            output_stat = output_band_stats[stat]
+            assert output_stat == pytest.approx(gt_value, rel=tolerance)
+
+def calculate_band_statistics(hypercube: xr.Dataset) -> dict:
+    """
+    Calculate statistics for each variable in the output cube.
+    
+    Parameters:
+        output_cube (xarray.Dataset): Input hypercube obtained through opening a netCDF file using xarray.Dataset.
+        
+    Returns:
+        dict: A dictionary containing statistics for each variable in the output cube. Keys are variable names 
+              (matching the pattern 'B01, B02, ...') and values are dictionaries containing mean, variance, min, max, 
+              and quantile statistics.
+    """
+
+    statistics = {}
+    band_names = [band_name for band_name in hypercube.data_vars\
+                      if re.match(r'^B\d',band_name)]
```

Comment:
This seems pretty inflexible, why only looking at bands that start with "B"? This is only the case for a subset of SENTINEL2_L2A bands